backend/controllers/passController.js

```diff
@@ -55,14 +62,47 @@ const viewPasses = asyncHandler(async (req, res) => {
         message: "Passes fetched successfully",
       })
     } else if (passes.length < 1) {
-      res.status(201).json({
-        success: true,
-        message: "No passes found",
-      })
+      throw new Error("No passes found")
     }
   } catch (err) {
     console.log(err)
+    res.status(400).json(err?.message)
+  }
+})
+
+//@desc Verify pass by gaurd
+//@route PUT /users/verifyPass
+const verifyPass = asyncHandler(async (req, res) => {
+  try {
+    await connectDB()
+    const user = await User.findById(req.user._id)
+    console.log(req.params._id)
+
+    if (user.role == "gaurd") {
```

Comment:
Same this logic can be moved to authorization middleware.

Reply:
done